lazyscribe/artifacts/joblib.py

```diff
@@ -58,31 +66,49 @@ def construct(
             the name of the artifact and the suffix for the class.
         created_at : datetime, optional (default None)
             When the artifact was created. If not supplied, :py:meth:`datetime.now` will be used.
-        **kwargs : Dict
+        package: str, optional (default None)
+            The package name or root module of the serializable python object.
+            Note: this may be different from the distribution name. e.g ``scikit-learn`` is
+            a distribution name, where as ``sklearn`` is the corresponding package name.
+        writer_kwargs : Dict, optional (default None)
             Keyword arguments for writing an artifact to the filesystem. Provided when an artifact
-            is logged to an experiment
+            is logged to an experiment.
+        **kwargs : Dict
+            Other keyword arguments. 
+            Usually class attributes obtained from a project JSON.
         """
+
+        if not package:
+            if not value:
+                raise ValueError(
+                    "If no ``package`` is specified, you must supply a ``value``."
+                )
+            package = value.__module__.split(".")[0]
+
+        try:
+            distribution = packages_distributions()[package][0]
+        except KeyError:
+            raise ValueError(f"{package} was not found.")
+
         try:
             import joblib
-            import sklearn
         except ImportError:
-            raise RuntimeError(
-                "Please install ``scikit-learn`` and ``joblib`` to use this handler."
-            )
+            raise RuntimeError(f"Please install ``joblib`` to use this handler.")
 
         return cls(
             name=name,
             value=value,
             fname=fname or f"{slugify(name)}.{cls.suffix}",
             created_at=created_at or datetime.now(),
-            writer_kwargs=kwargs,
-            sklearn_version=sklearn.__version__,
-            joblib_version=joblib.__version__,
+            writer_kwargs=writer_kwargs or {},
+            package=package,
+            package_version=kwargs.get("package_version") or version(distribution),
+            joblib_version=kwargs.get("joblib_version") or joblib.__version__,
```

Comment:
```suggestion
            package_version=kwargs.get("package_version", version(distribution)),
            joblib_version=kwargs.get("joblib_version", joblib.__version__),
```

Reply:
Good catch. That was dumb. Side effects of late night coding lol